jetty-util/src/main/java/org/eclipse/jetty/util/thread/AutoLock.java

```diff
@@ -64,4 +119,69 @@ public void close()
     {
         _lock.unlock();
     }
+
+    /**
+     * <p>A reentrant lock with a condition that can be used in a try-with-resources statement.</p>
+     * <p>Typical usage:</p>
+     * <pre>
+     * // Waiting
+     * try (AutoLock lock = _lock.lock())
+     * {
+     *     lock.await();
+     * }
+     *
+     * // Signaling
+     * try (AutoLock lock = _lock.lock())
+     * {
+     *     lock.signalAll();
+     * }
+     * </pre>
+     */
+    public static class WithCondition extends AutoLock
+    {
+        private final Condition _condition = newCondition();
```

Comment:
I don't like this new method as it allocates a new Condition on every lock??  So at the very least that is garbage.... but does it even work so one locker is going to have a different condition to another?   I think this is just wrong?

If you want to avoid the extra field to hold a condition then you need a class `AutoLockWithCondition`... and maybe `AutoLockWithBiCondition`

Reply:
Ah ignore this... I misread the code.
Still not a fan of this though.